streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -584,6 +579,7 @@ public boolean test(final K1 key, final V1 value) {
         String name = topology.newName(LEFTJOIN_NAME);
 
         topology.addProcessor(name, new KStreamKTableLeftJoin<>((KTableImpl<K, ?, V1>) other, joiner), this.name);
+        topology.connectProcessorAndStateStores(name, other.getStoreName());
```

Comment:
@mjsax For example:

```
table1 = builder.table("topic1", "store1");
table2 = builder.table("topic2", "store2");
table3 = table1.join(table2, ...);
stream2 = stream1.join(table3);
```

In this case, `table3` is not materialized, and in `KTableKTableJoinMerger` you can see that it is actually trying to access both `table1` and `table2`'s materialized store, `store1` and `store2`, as indicated in `KTableKTableJoinValueGetter`. So I think in this case we will still hit a `TopologyException` (feel free to double check if that is the case).